include/real/integer_number.hpp

```diff
@@ -77,113 +77,111 @@ namespace boost{
 			 * This function will simply Subtract vector A from B, and A should be bigger than B.
 			 * These conditions will be checked by operator overload and this function will be called in an appropriate manner.
 			 **/
-			integer<T> subtract_integer_number(integer<T> &other)
+			integer_number<T> subtract_integer_number(integer_number<T> &other)
 			{
 				// it is assumed that digits in (*this) are greater than digits in other
 				int borrow = 0;
 				T val = 0;
-				size_t n1 = (*this).digits.size();
+				size_t n1 = digits.size();
 				size_t n2 = other.digits.size();
 				std::vector<T> result_vec(n1);
-				for(int i=0;i<n1;i++)
+				for(int i = 0; i < n1; ++i)
 				{
 					val += borrow;
 					borrow = 0;
-					if(i<n2) val += other.digits[n2-i-1];
-					if(val > (*this).digits[n1-i-1])
+					if(i < n2) val += other.digits[n2 - i - 1];
+					if(val > digits[n1 - i - 1])
 					{
 						borrow = 1;
 						// equivqlent to (*this).digits[n1-i] + BASE - val
-						result_vec[n1-i-1] = (BASE - val) + (*this).digits[n1-i-1];
+						result_vec[n1 - i - 1] = (BASE - val) + digits[n1 - i - 1];
 					}
-					else result_vec[n1-i-1] = (*this).digits[n1-i-1] - val;
+					else result_vec[n1 - i - 1] = digits[n1 - i - 1] - val;
 					val = 0;
 				}
 				// this function always expects (*this) to bigger than other, so if borrow is not zero at the end
 				// there should be error
-				if(borrow!=0); 
+				if(borrow != 0); 
 					//throw subtraction_number_required_lower_exception();
 
 				// now we need to remove extra zeroes, that we may get after subtraction
 
 				typename std::vector<T> :: iterator itr = result_vec.begin();
-				while((*itr)==0 && itr!=result_vec.end()) itr++;
+				while((*itr) == 0 && itr != result_vec.end()) ++itr;
 				// if initial element was zero, then we need to delete all initial zeroes
-				if(result_vec[0]==0)
+				if(result_vec[0] == 0)
```

Comment:
Do you need to really erase them, maybe just keeping track on a variable of how many are not usefull is enough. I worried if we do many subtractions we going to have a lot of memory reallocations by erasing elements front of a vector.

Reply:
It may be a premature optimization, but lets keep in mind that is something we can do if we find it relevant.

Reply:
So, for now, should I keep it the way it is or change it?